ipn/ipnlocal/peerapi.go

```diff
@@ -542,7 +543,30 @@ func (h *peerAPIHandler) logf(format string, a ...any) {
 	h.ps.b.logf("peerapi: "+format, a...)
 }
 
+// validPeerAPIRequest reports whether r is a valid PeerAPI request.
+func (h *peerAPIHandler) validPeerAPIRequest(r *http.Request) bool {
+	if r.Referer() != "" {
+		return false
+	}
+	if r.Header.Get("Origin") != "" {
+		return false
+	}
+	if r.Host == "peer" {
```

Comment:
I would put this logic in a helper e.g. `validHost`. Then if we add more tests later on, this early return true doesn't end up skipping them.

Reply:
good idea, done.

Reply:
in the signature I sent you on chat, it was one method but it returned an error so tests could validate the expected error path fired

Reply:
Made it return an error instead. PTAL

Reply:
but this part doesn't return an error. it can fail for two reasons.

just move this back into the other method and return an error in all bad paths?

Reply:
moving it back reduces readability when combined with "Then if we add more tests later on, this early return true doesn't end up skipping them." I made `validateHost` also return an error so that it's clear what the error is.